sdks/go/pkg/beam/io/fhirio/utils_test.go

```diff
@@ -118,19 +127,23 @@ func (m *fakeReaderCloser) Read(b []byte) (int, error) {
 	return m.fakeRead(b)
 }
 
-func validateCounter(pipelineResult beam.PipelineResult, expectedCounterName string, expectedCount int) error {
-	counterResults := pipelineResult.Metrics().AllMetrics().Counters()
-	if len(counterResults) != 1 {
-		return fmt.Errorf("counterResults got length %v, expected %v", len(counterResults), 1)
+func validateCounter(t *testing.T, pipelineResult beam.PipelineResult, expectedCounterName string, expectedCount int) {
+	t.Helper()
+
+	counterResults := pipelineResult.Metrics().Query(func(mr beam.MetricResult) bool {
+		return mr.Name() == expectedCounterName
+	}).Counters()
+
+	if expectedCount == 0 && len(counterResults) == 0 {
+		return
 	}
-	counterResult := counterResults[0]
 
-	if counterResult.Name() != expectedCounterName {
-		return fmt.Errorf("counterResult.Name() is '%v', expected '%v'", counterResult.Name(), expectedCounterName)
+	if len(counterResults) != 1 {
+		t.Fatalf("counter %v got length %v, expected %v", expectedCounterName, len(counterResults), 1)
```

Comment:
It looks like this error string should be something like `got %v counters with name %v, expected 1`, correct? Am I understanding this? The error message kind of looks like you're saying we got the wrong results for the counter.